homeassistant/components/daikin.py

```diff
@@ -0,0 +1,144 @@
+"""
+Platform for the Daikin AC.
+
+For more details about this component, please refer to the documentation
+https://home-assistant.io/components/daikin/
+"""
+import logging
+import time
+from socket import timeout
+from threading import Lock
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.discovery import SERVICE_DAIKIN
+from homeassistant.helpers import discovery
+from homeassistant.helpers.discovery import load_platform
+from homeassistant.const import (
+    CONF_HOSTS, CONF_ICON, CONF_NAME,
+    CONF_MONITORED_CONDITIONS
+)
+
+REQUIREMENTS = ['pydaikin==0.4']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'daikin'
+HTTP_RESOURCES = ['aircon/get_sensor_info', 'aircon/get_control_info']
+
+ATTR_TARGET_TEMPERATURE = 'target_temperature'
+ATTR_INSIDE_TEMPERATURE = 'inside_temperature'
+ATTR_OUTSIDE_TEMPERATURE = 'outside_temperature'
+
+# default scan interval in seconds
+DEFAULT_SCAN_INTERVAL = 60
+
+COMPONENT_TYPES = ['climate', 'sensor']
+
+SENSOR_TYPES = {
+    ATTR_INSIDE_TEMPERATURE: {
+        CONF_NAME: 'Inside Temperature',
+        CONF_ICON: 'mdi:thermometer'
+    },
+    ATTR_OUTSIDE_TEMPERATURE: {
+        CONF_NAME: 'Outside Temperature',
+        CONF_ICON: 'mdi:thermometer'
+    }
+
+}
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Optional(CONF_HOSTS, default=[]): vol.Schema([cv.string]),
+        vol.Optional(
+            CONF_MONITORED_CONDITIONS,
+            default=list(SENSOR_TYPES.keys())
+        ): vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)])
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
+    """Establish connection with Daikin."""
+    def discovery_dispatch(service, discovery_info):
+        """Dispatcher for Daikin discovery events."""
+        host = discovery_info.get('ip')
+
+        if daikin_api_setup(hass, host) is None:
+            return
+
+        for component in COMPONENT_TYPES:
+            load_platform(hass, component, DOMAIN, discovery_info,
+                          config)
+
+    discovery.listen(hass, SERVICE_DAIKIN, discovery_dispatch)
+
+    for host in config.get(DOMAIN, {}).get(CONF_HOSTS, []):
+        if daikin_api_setup(hass, host) is not None:
+            discovery_info = {
+                'ip': host,
+                CONF_MONITORED_CONDITIONS:
+                    config.get(DOMAIN, {}).get(CONF_MONITORED_CONDITIONS)
```

Comment:
Simplify this since at this point you know that you're setting up from config and the config validation will have put defaults in place.
```py
config[DOMAIN][CONF_MONITORED_CONDITIONS]
```